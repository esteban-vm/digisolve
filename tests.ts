import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
