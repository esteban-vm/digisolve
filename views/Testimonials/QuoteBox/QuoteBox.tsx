import type { FunctionalComponent } from '@/types'
import Image, { type StaticImageData } from 'next/image'
import { styled } from '@/utils/styles'

/** Properties for quote box component */
export type QuoteProps = {
  /** The client's name */
  name: string
  /** The client's photo */
  photo: StaticImageData
}

const QuoteBoxComponent: FunctionalComponent<QuoteProps> = ({ name, photo, ...rest }) => {
  return (
    <div {...rest}>
      <blockquote>Lorem ipsum dolor, sit amet consectetur adipisicing elit. At, praesentium.</blockquote>
      <cite>
        <Image src={photo} alt={name} />
        {name}
      </cite>
    </div>
  )
}

const QuoteBox = styled(QuoteBoxComponent)`
  text-align: left;
  margin-bottom: 50px;

  blockquote {
    padding: 2%;
    font-style: italic;
    line-height: 145%;
    position: relative;
    margin-top: 40px;

    ::before {
      content: '\\201C';
      display: block;
      font-size: 500%;
      position: absolute;
      left: -5px;
      top: -5px;
    }
  }

  cite {
    display: block;
    margin-top: 25px;
    font-size: 90%;

    img {
      border-radius: 50%;
      height: 20%;
      vertical-align: middle;
      margin-right: 10px;
    }
  }
`

QuoteBox.defaultProps = {
  className: 'col span_1_of_3',
}

export default QuoteBox
