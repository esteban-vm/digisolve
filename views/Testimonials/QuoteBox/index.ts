export { default, type QuoteProps } from './QuoteBox'
