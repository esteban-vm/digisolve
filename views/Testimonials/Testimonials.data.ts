import type { PropsWithImage } from '@/types'
import type { TestimonialProps } from './Testimonial'

const text = 'Voluptates hic pariatur reiciendis minus vel labore dolorum.'
const width = 72
const height = 72

const testimonials: PropsWithImage<TestimonialProps>[] = [
  {
    id: 'c2f0fc36-aad3-4e98-9ff8-61a30257d41c',
    customer: 'Herman Gibson',
    text,
    image: { src: '/img/client-1.jpg', width, height },
  },
  {
    id: 'efc0be82-fe34-4df9-8821-cf195a75da1d',
    customer: 'Peter Kemmer',
    text,
    image: { src: '/img/client-2.jpg', width, height },
  },
  {
    id: 'c679bf1c-1387-4aa0-9eec-ab2d28d79310',
    customer: 'Naomi Stoltenberg',
    text,
    image: { src: '/img/client-3.jpg', width, height },
  },
]

export default testimonials
