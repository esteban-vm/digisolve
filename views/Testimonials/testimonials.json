[
  {
    "image": "/img/client-1.jpg",
    "text": "Aaron Cruz",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-2.jpg",
    "text": "Jeff Jones",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-3.jpg",
    "text": "Helen Smith",
    "height": 72,
    "width": 72
  }
]
