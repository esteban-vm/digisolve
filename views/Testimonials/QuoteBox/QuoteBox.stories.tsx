import type { Meta, StoryObj } from '@storybook/react'
import { within, userEvent } from '@storybook/testing-library'
import { styled } from '@/styles'
import { quotes } from '../Testimonials'
import QuoteBox from './QuoteBox'

const Wrapper = styled.div`
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 75vw;

  article {
    background-color: var(--color-light);
    padding: 1rem;
  }
`

export default {
  argTypes: {
    photo: {
      control: false,
    },
    onMouseEnter: {
      action: true,
      control: false,
    },
  },
  component: QuoteBox,
  title: 'Views/Testimonials/Quotes',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Story {...context} />
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const photo = canvas.getByRole('img')
    userEvent.hover(photo)
  },
} as Meta<typeof QuoteBox>

const [quote1, quote2, quote3] = quotes

export const QuoteExample1: StoryObj<typeof QuoteBox> = { args: quote1 }
export const QuoteExample2: StoryObj<typeof QuoteBox> = { args: quote2 }
export const QuoteExample3: StoryObj<typeof QuoteBox> = { args: quote3 }
