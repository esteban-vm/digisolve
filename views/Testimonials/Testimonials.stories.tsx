import type { Meta } from '@storybook/react'
import Testimonials from './Testimonials'

export default {
  component: Testimonials,
  title: 'Views/Testimonials',
} as Meta<typeof Testimonials>

export const TestimonialsView = () => <Testimonials />
