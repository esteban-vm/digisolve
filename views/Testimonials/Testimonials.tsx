import type { FunctionalComponent } from '@/types'
import { client1, client2, client3 } from '@/utils/images'
import { styled } from '@/utils/styles'
import QuoteBox, { type QuoteProps } from './QuoteBox'

export const quotes: QuoteProps[] = [
  { name: 'Aaron Cruz', photo: client1 },
  { name: 'Jeff Jones', photo: client2 },
  { name: 'Helen Smith', photo: client3 },
]

const TestimonialsComponent: FunctionalComponent = (props) => {
  return (
    <section {...props}>
      <div className='row'>
        <h2>Our Testimonials</h2>
      </div>
      <div className='row'>
        {quotes.map((quote, index) => (
          <QuoteBox key={index} {...quote} role='listitem' />
        ))}
      </div>
    </section>
  )
}

const Testimonials = styled(TestimonialsComponent)`
  background-image: linear-gradient(var(--color-overlay3), var(--color-overlay4)), url('/img/testimonial-bg.jpg');
  background-size: cover;
  background-attachment: fixed;
  text-align: center;
  color: var(--color-white);

  h2 {
    font-size: 250%;
    margin-bottom: 20px;
    margin-top: 150px;
  }
`

export default Testimonials
