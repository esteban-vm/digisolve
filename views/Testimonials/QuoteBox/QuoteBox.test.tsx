import { render, cleanup, screen, create } from '@/tests'
import { quotes } from '../Testimonials'
import QuoteBox from './QuoteBox'

describe('ðŸ§ª QUOTE BOX:', () => {
  const [testQuote] = quotes

  describe('display tests:', () => {
    let parent: HTMLElement

    beforeEach(() => {
      render(<QuoteBox {...testQuote} />)
      parent = screen.getByRole('article')
    })

    afterEach(cleanup)

    it('should display the photo', () => {
      const photo = screen.getByRole('img', { name: testQuote.name })
      expect(photo).toBeInTheDocument()
      expect(photo).toBeVisible()
      expect(parent).toContainElement(photo)
    })

    it('should display the name', () => {
      const name = screen.getByText(testQuote.name)
      expect(name).toBeInTheDocument()
      expect(name).toBeVisible()
      expect(parent).toContainElement(name)
    })

    it('should display the quote', () => {
      const quote = screen.getByText(/^lorem ipsum/i)
      expect(quote).toBeInTheDocument()
      expect(quote).toBeVisible()
      expect(parent).toContainElement(quote)
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<QuoteBox {...testQuote} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
