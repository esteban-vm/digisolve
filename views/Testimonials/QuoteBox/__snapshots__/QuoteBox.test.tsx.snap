// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª QUOTE BOX: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: left;
  margin-bottom: 50px;
}

.emotion-0 blockquote {
  padding: 2%;
  font-style: italic;
  line-height: 145%;
  position: relative;
  margin-top: 40px;
}

.emotion-0 blockquote::before {
  content: '\\201C';
  display: block;
  font-size: 500%;
  position: absolute;
  left: -5px;
  top: -5px;
}

.emotion-0 cite {
  display: block;
  margin-top: 25px;
  font-size: 90%;
}

.emotion-0 cite img {
  border-radius: 50%;
  height: 20%;
  vertical-align: middle;
  margin-right: 10px;
}

<article
  className="col span_1_of_3 emotion-0"
>
  <blockquote>
    Lorem ipsum dolor, sit amet consectetur adipisicing elit. At, praesentium.
  </blockquote>
  <cite>
    <img
      alt="Aaron Cruz"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={40}
    />
    Aaron Cruz
  </cite>
</article>
`;
